src/System.Drawing.Common/src/System/Drawing/Brush.cs

```diff
@@ -100,34 +60,13 @@ protected virtual void Dispose(bool disposing)
                     Debug.Assert(status == SafeNativeMethods.Gdip.Ok, "GDI+ returned an error status: " + status.ToString(CultureInfo.InvariantCulture));
 #endif
                 }
-                catch (Exception ex)
-                {
-                    if (ClientUtils.IsSecurityOrCriticalException(ex))
-                    {
-                        throw;
-                    }
-
-                    Debug.Fail("Exception thrown during Dispose: " + ex.ToString());
-                }
```

Comment:
> Do you reckon I should revert this?

Yes.

> I didn't quite understand your comment about mono.

The Mono runtime may behave differently in various situations from CoreCLR, so your assertions about behaviors of CoreCLR and what exceptions the runtime may or may not throw (e.g. AccessViolationException) may or may not hold if this code is used on Mono.  And the plan is for more and more CoreFx code to be used to replace code in Mono's libraries.